atsign-foundation/at_client_sdk
# feat: at_client put method - introduce shouldEncrypt flag

Fixes https://github.com/atsign-foundation/at_client_sdk/issues/1401
Depends on https://github.com/atsign-foundation/at_server/pull/2099 for intermittent functional test failure in sync_multiple_client_test

**- What I did**
- Ability for clients to set shouldEncrypt=false when they don't want to encrypt data for self and shared key

**- How I did it**
- added a flag shouldEncrypt with default value=true in PutRequestOptions. This value will be ignored for public keys(metadata.isEncrypted is always false). For self and sharedkey this value will be honoured and data will be encrypted(shouldEncrypt=true) or not encrypted(shouldEncrypt=false)
- Refactored put_request_transformer for readability
- Changes in get_response_transformer to handle two scenarios  1) isEncrypted for new data(post these changes where isEncrypted=true for self/shared keys) 2) isEncrypted=false for old self/shared key.
- modified sync_multiple_client_test to replicate the intermittent issue consistently
- added unit and functional test

**- How to verify it**
- tests should pass
Dependency tests
https://github.com/atsign-foundation/at_libraries/pull/673
https://github.com/atsign-foundation/noports/pull/1387
TODO:
Replace dependency overrides with published versions
at_commons PR to be reviewed and merged 
https://github.com/atsign-foundation/at_libraries/pull/669
